### ISSUE TITLE
Validator class workflows should be mentioned in usage docs

### ISSUE THREAD

**Issue by cdunklau** [python-jsonschema/jsonschema#590]
I was looking for a way to do what `jsonschema.validators.validator_for` does, but didn't find it in the initial "basic usage" section. A band-aid could be to expand the basic usage documentation (say, where it points out "If you know you have a valid schema already...").

...however I must say I was confused by the layout of the docs. The table of contents seems to imply a more narrative style, but the docs intersperse prose with generated API documentation. I think a cleaner separation of narrative and API docs (with appropriate cross-references) could be more friendly and effective.

**Comment by Julian** [comment]
Ages later, there's still poor separation (API docs interspersed in narrative docs) but that's more now a question of writing more/better narrative docs, as we do indeed now have a separate API reference [here](https://python-jsonschema.readthedocs.io/en/stable/api/) and also mention validator classes in the docs for validate.

TL;DR: docs still wonky in places and need improving, but this specific issue may be covered now. Feedback or help of course very welcome.
